### PR TITLE
Bug 1347721 - Add Mgmt Command to delete orphaned FailureLine records

### PR DESCRIPTION
**Pull request by camd**
This addresses [Bug 1347721](https://bugzilla.mozilla.org/show_bug.cgi?id=1347721)

I added a few extra params to control which items to run.  I have this running on Prototype in one of the Heroku Schedulers.  I was hitting mem issues, so I am now using a smaller chunk size and 2X dyno, which seems to be working well.

I'm happy to take another approach, if that makes more sense.  :)

### CODE REVIEW

File: treeherder/model/management/commands/cycle_data.py [hunk 1]
```diff
@@ -55,32 +72,65 @@ def handle(self, *args, **options):
...
+                break
 
-        self.cycle_non_job_data(options['chunk_size'], options['sleep_time'])
+            FailureLine.objects.filter(id__in=old_fline_ids).delete()
+            self.debug("FailureLines deleted, sleeping for {}".format(sleep_time))
+            time.sleep(sleep_time)
 
-    def cycle_non_job_data(self, chunk_size, sleep_time):
+    def cycle_non_job_data(self, chunk_size, sleep_time, cycle_interval):
+        self.debug("delete unused JobType records")
```
Comment (edmorley) [inline treeherder/model/management/commands/cycle_data.py]: Maybe s/delete/deleting/ ?

File: treeherder/model/management/commands/cycle_data.py [hunk 2]
```diff
@@ -55,32 +72,65 @@ def handle(self, *args, **options):
...
+
+        if options['target'] in ["all", "refdata"]:
+            self.cycle_non_job_data(
+                options['chunk_size'],
+                options['sleep_time'],
+                cycle_interval)
+
+    def cycle_expired_log_records(self, chunk_size, sleep_time, iterations, cycle_interval):
+        self.debug("delete expired FailureLines")
+        for x in range(0, iterations):
```
Comment (edmorley) [inline treeherder/model/management/commands/cycle_data.py]: I think maybe we can remove the iterations here and instead run until completion? (Now I've run the manual purge, there will be much less to delete)

File: treeherder/model/management/commands/cycle_data.py [hunk 3]
```diff
@@ -55,32 +72,65 @@ def handle(self, *args, **options):
...
+            old_fline_ids = list(FailureLine.objects.filter(
```
Comment (edmorley) [inline treeherder/model/management/commands/cycle_data.py]: Casting to a list will cause queryset evaluation and increase memory usage, can we avoid that? Can the `.delete()` be combined with this?

File: treeherder/model/management/commands/cycle_data.py [hunk 4]
```diff
@@ -22,6 +24,13 @@ def add_arguments(self, parser):
             default=False,
             help='Write debug messages to stdout'
         )
+        parser.add_argument(
+            '--target',
```
Comment (edmorley) [inline treeherder/model/management/commands/cycle_data.py]: What do you think about "type" as an alternative to "target"? (I don't mind either way :))

**Comment by camd** [comment]
``cycle_data`` is now properly cleaning this table.  I can no longer find any orphaned records.  So we don't need this PR any more.

**Pull request closed without merging**